ilastik/applets/batchProcessing/batchProcessingApplet.py

```diff
@@ -54,11 +56,42 @@ def parse_known_cmdline_args(self, cmdline_args):
             help="Distributed mode. Used for running ilastik on HPCs via SLURM/srun/mpirun",
             action="store_true",
         )
+
+        default_block_roi = Slice5D.all(x=slice(0, 256), y=slice(0, 256), z=slice(0, 256), t=slice(0, 1))
+
+        def parse_distributed_block_roi(value: str) -> Slice5D:
+            parsed = ast.literal_eval(value)
+            if isinstance(parsed, dict):
+                if not set(parsed.keys()).issubset("xyztc"):
+                    raise ValueError(f"Bad keys for distributed-block-roi: {value}")
+                if not all(isinstance(v, (int, None.__class__)) for v in parsed.values()):
+                    raise ValueError(f"Bad values for distributed-block-roi: {value}")
+                overrides = {k: slice(0, int(v)) for k, v in parsed.items()}
+            elif isinstance(parsed, int):
+                overrides = {k: slice(0, parsed) for k in "xyz"}
+            else:
+                raise TypeError(f"Could not convert value {value} into a Slice5D")
+
+            return Slice5D(**{**default_block_roi.to_dict(), **overrides})
+
         parser.add_argument(
-            "--distributed_block_size",
-            help="The length of the side of the tiles used in distributed mode",
-            type=int,
-            default=256,
+            "--distributed_block_roi",
+            "--distributed-block-roi",
+            help=textwrap.dedent(
+                """
+                Determines the dimensions of the blocks used to split the data in distributed mode.
+                Values can be:"
+                    A literal python Dict[str, Optional[int]], with keys in 'xyztc'. Missing keys will default like so:
```

Comment:
Nice, making this a dict frees us from all the ambiguities of passing just a sequence! Being very nitpicky I'd mention the simple value for block roi size first. This is what most users want. Typing a dictionary seems like pretty advanced usage without clear benefits. I most cases that I can think of it is more beneficial to crank up the spacial dimensions.

Reply:
> Being very nitpicky I'd mention the simple value for block roi size first

OK. I think I liked the dict explanation first, so that the simple value can then be explained as a special case of the dict, but I do agree that it might overcomplicate for normal users.

> Also our command line parser is very very picky about having spaces in values.

Are you referring to [this](https://github.com/ilastik/ilastik/blob/a1bb868b0a8d43ac3c89e681cc89d43be9591ff7/ilastik/utility/commandLineProcessing.py#L68)? If so, then we're safe; I'm using `ast.literal_eval` for the parsing. Other than that, there's just the responsibility of using quotes when passing parameters in the command line, and `argparse` should handle it just fine.

Reply:
hm, I was more referring to a [bug in cpython](https://bugs.python.org/issue22433)

Reply:
Ha. I wasn't aware of that.

Anyway, AFAIU, it only happens when there's an "unknown optional" and a "known positional", which causes the "known positional" to capture the "unknown optional". Since the `BatchProcessingApplet` extends the `DataSelectionApplet` argument parser with the `--distributed-block-roi` parameter, by the time the `unspecified_input_files` positional argument is parsed, `--distributed-block-roi` will no longer be unknown, so the parsing won't break. And since the workflows themselves don't handle any other positional arguments by themselves, we shouldn't get this miss-parsing when they run their `parse_known_args` either.

It's too bad that there is now this extra bit of implicit dependency in the argument parsing. Maybe we should start deprecating the usage of the positional arguments and eventually remove it altogether. Anyway, I added another variation to the tests for the sake of paranoia.

Reply:
I agree that we need to phase positional arguments out. They do more harm than they add convenience imo.